rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -1024,23 +997,21 @@ def _prepare_layers(self) -> None:
                 self.config[C_EMB],
                 self.config[SCALE_LOSS],
             )
-            self.train_metrics["i_loss"] = tf.keras.metrics.Mean(name="i_loss")
-            self.train_metrics["i_acc"] = tf.keras.metrics.Mean(name="i_acc")
-            self.eval_metrics["val_i_loss"] = tf.keras.metrics.Mean(name="val_i_loss")
-            self.eval_metrics["val_i_acc"] = tf.keras.metrics.Mean(name="val_i_acc")
         else:
             self._loss_label = None
 
+    def _prepare_entity_recognition_layers(self):
         self._crf = None
         if self.config[ENTITY_RECOGNITION]:
             self._embed["logits"] = tf_layers.Embed(
                 self._num_tags, self.config[C2], "logits"
             )
             self._crf = tf_layers.CRF(self._num_tags, self.config[C2])
-            self.train_metrics["e_loss"] = tf.keras.metrics.Mean(name="e_loss")
-            self.train_metrics["e_f1"] = tf.keras.metrics.Mean(name="e_f1")
-            self.eval_metrics["val_e_loss"] = tf.keras.metrics.Mean(name="val_e_loss")
-            self.eval_metrics["val_e_f1"] = tf.keras.metrics.Mean(name="val_e_f1")
+
+            self.metric_f1_score = tfa.metrics.F1Score(
```

Comment:
```suggestion
            self._crf_f1_score = tfa.metrics.F1Score(
```
we create for num_tags, so cannot use for anything else

Reply:
we don't actually need to collect this metric, so it can be part of the dict

Reply:
there is no dict anymore, we can remove `TraningMetrics` once we adapted core

Reply:
by dict, I mean my suggestion about putting all the layers into a dict

Reply:
ah ok, will try that